pkg/services/ngalert/eval/eval.go

```diff
@@ -155,11 +162,57 @@ func executeCondition(ctx AlertExecCtx, c *models.Condition, now time.Time, data
 		return ExecutionResults{Error: err}
 	}
 
+	// eval captures for the '__value__' label.
+	captures := []NumberValueCapture{}
+
+	captureVal := func(refID string, labels data.Labels, value *float64) {
+		captures = append(captures, NumberValueCapture{
+			Var:    refID,
+			Value:  value,
+			Labels: labels.Copy(),
+		})
+	}
+
 	for refID, res := range execResp.Responses {
-		if refID != c.Condition {
-			continue
+		// for each frame within each response, capture value if it is a number.
+		for _, frame := range res.Frames {
+			if len(frame.Fields) != 1 || frame.Fields[0].Type() != data.FieldTypeNullableFloat64 {
+				continue
+			}
+			var v *float64
+			if frame.Fields[0].Len() == 1 {
+				v = frame.At(0, 0).(*float64) // type checked above
+			}
+			captureVal(frame.RefID, frame.Fields[0].Labels, v)
+		}
+
+		if refID == c.Condition {
+			result.Results = res.Frames
+		}
+	}
+
+	// add capture values as data frame metadata to each result (frame) that has matching labels.
+	for _, frame := range result.Results {
+		if frame.Meta != nil && frame.Meta.Custom != nil {
+			if _, ok := frame.Meta.Custom.([]classic.EvalMatch); ok {
+				continue // do not overwrite EvalMatch from classic condition.
+			}
+		}
+
+		frame.SetMeta(&data.FrameMeta{}) // overwrite metadata
+
+		if len(frame.Fields) == 1 {
```

Comment:
What happens in other cases? Why don't we attach the data for those?

Reply:
It would be a time series, or perhaps empty, or something unexpected.